src/App.tsx

```diff
@@ -1,22 +1,80 @@
-import React from "react";
+import React, { useState } from "react";
 import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import {
+  createBrowserRouter,
+  Navigate,
+  RouterProvider,
+} from "react-router-dom";
 
 import styles from "./App.module.scss";
 import { MovieListPage } from "./pages";
-import { MovieProvider } from "@context/MovieContext";
+import { MovieDetails } from "@components/MovieDetails/MovieDetails";
+import { SearchForm } from "@components/SearchForm/SearchForm";
+import { Button, ButtonTexts, Header } from "@shared/components";
+import { AddEditMovieDialog } from "@components/AddEditMovieDialog/AddEditMovieDialog";
+import { IMovieInfo } from "@components/MovieList/MovieCard/MovieCard.types";
+import { RoutePaths } from "./App.types";
+import { getMovieById } from "./api/fetchData";
 
-const { appContainer } = styles;
+const { appContainer, addMovieButton } = styles;
 
 const queryClient = new QueryClient();
 
-const App = () => (
-  <QueryClientProvider client={queryClient}>
-    <div className={appContainer} id="main-page">
-      <MovieProvider>
-        <MovieListPage />
-      </MovieProvider>
-    </div>
-  </QueryClientProvider>
-);
+const App = () => {
+  const [isAddMovieModalOpened, setIsAddMovieModalOpened] =
+    useState<boolean>(false);
+
+  const router = createBrowserRouter([
+    {
+      path: RoutePaths.Home,
+      element: <MovieListPage />,
+      children: [
+        {
+          index: true,
+          element: (
+            <Header>
+              <SearchForm />
+              <Button
+                className={addMovieButton}
+                buttonText={ButtonTexts.AddMovie}
+                onClick={() => setIsAddMovieModalOpened(true)}
+              />
+            </Header>
+          ),
+        },
+        {
+          path: ":movieId",
+          element: <MovieDetails />,
+          loader: async ({ params }): Promise<IMovieInfo> =>
+            getMovieById(params.movieId as string),
+        },
+      ],
+    },
+    {
+      path: "*",
+      element: <Navigate to={RoutePaths.Home} replace />,
+    },
+  ]);
+
+  const handleAddMovie = (moviePayload: IMovieInfo): void => {
+    console.log("New movie added:", moviePayload);
+    setIsAddMovieModalOpened(false);
+  };
+
+  return (
+    <QueryClientProvider client={queryClient}>
+      <div className={appContainer} id="main-page">
+        <RouterProvider router={router} />
+      </div>
+
+      {isAddMovieModalOpened && (
+        <AddEditMovieDialog
+          onSubmit={handleAddMovie}
+          onCancel={() => setIsAddMovieModalOpened(false)}
+        />
+      )}
+    </QueryClientProvider>
+  );
+};
```

Comment:
From time to time this component
![image](https://github.com/user-attachments/assets/aeeed454-cc87-4c0e-8e28-2ea48a305b22)

disappear. When you switched between genres, when you switch sortBy. Try to fix it. React should rerenders only numbers of the result.

Reply:
Indeed, fixed, thx